### DISCUSSION TITLE
Table of pore radii

### DISCUSSION THREAD

**Discussion by apvij** [PMEAL/porespy#683] [Q&A]
Hello, 
I am using local thickness on 2d images, and need to be able to access the radius length data in order to do some statistical analysis and combine some datasets. Is there a way I can get a list or table of the radii? I am essentially using this tutorial exactly. 
https://porespy.org/examples/filters/tutorials/local_thickness.html
I just adjusted the voxel size for my image. 
Thanks for your help.

**Comment by jgostick** [comment]
Hi @apvij, it depends what you mean by radii?  The Results object returned from the ``psd = ps.metrics.pore_size_distribution(im=thk)`` function is basically tabulated data, but it's for data of the histogram.  The attributes of ``psd`` that are used to make the plot can just as easily be output to terminal as a table.  

If really want a list of ALL RADII in the image, you can just do ``r = thk[im]``, which will pull out all the local thickness values where ``im`` is ``True``. The result will be a flattened array (i.e. a 1D list).  This is not too use without some sort of 'count' of the number of each occurence though, which is waht the psd function does.

**Comment by apvij** [comment]
Can I get either a list of radii where if the same radius occurs more than once it is repeated in the list, or can I get a frequency for each radius rather than using psd?

**Reply by jgostick** [reply]
This just requires regular numpy array operations.  I'll leave you to play around with your image and I'm sure you'll figure it out.
